app/(tabs)/index.tsx

```diff
@@ -40,15 +41,64 @@ export default function TabOneScreen() {
 		} catch (error) {
 			console.warn(error);
 			Alert.alert('An error occurred');
-		} finally {
-			setLoading(false);
+		}
+	};
+
+	const getEventsWithFilters = async (categoryId?: number, searchText?: string, sortBy?: string) => {
+		try {
+			let fetchedEvents;
+			if (categoryId !== undefined) {
+				const headers = {
```

Comment:
Would be better to call it `const requestParams`.

Reply:
No